app/models/external_reference.rb

```diff
@@ -0,0 +1,11 @@
+class ExternalReference < ApplicationRecord
+  belongs_to :item, polymorphic: true
+  belongs_to :oauth_application, class_name: "Doorkeeper::Application"
+  belongs_to :territory
+
+  # Permet de rendre les créations d'objets idempotentes
+  validates :external_id, uniqueness: { scope: %i[item_type oauth_application_id territory] }
+
+  # Evite que le même item ai deux ids différents dans le même système
```

Comment:
Ce commentaire n’est pas clair pour moi

```suggestion
  # Evite que nous ayons 2 lignes liens différents pour le même item en provenance de la même application externe
```

Pas sur que ça soit mieux 😅